lib/matplotlib/legend_handler.py

```diff
@@ -316,7 +316,7 @@ def get_sizes(self, legend, orig_handle,
             numpoints = self.get_numpoints(legend)
             if numpoints < 4:
                 sizes = [.5 * (size_max + size_min), size_max,
```

Comment:
all the more reason to unit test it, no?
It is a very easy unit test to check that the return lengths is equal to numpoints.

Also, my two cents is that I would much prefer the returned size to be sorted (only matters for numpoints == 3)